### DISCUSSION TITLE
How to use RegisterDefaultPgType with connection Pool?

### DISCUSSION THREAD

**Discussion by stasvo** [jackc/pgx#1532] [Q&A] [answered]
As I understood type mappings can be registered with pgtype.Map.RegisterDefaultPgType.
And if I am not wrong in case of use *Conn (without pool) for register type possible to use this method: conn.TypeMap().RegisterDefaultPgType()
But in case of *pgxpool.Pool there no TypeMap() method, how possible to register type while use connection pool (*pgxpool.Pool)?

**Comment by jackc** [comment, marked as answer]
You can do type registration in an `AfterConnect` hook with pgxpool. See https://pkg.go.dev/github.com/jackc/pgx/v5@v5.3.1/pgxpool#Config.

**Reply by stasvo** [reply]
Thank you.
